scripts/reqs/idgen.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Copyright (C) 2025 Red Hat, Inc. Alessandro Carminati <acarmina@redhat.com>
+#
+# Scans a given directory tree and adds missing SPDX-Req-ID identifiers
```

Comment:
In the comments it would be good to show a quick command line example that shows how to use the script